provider/cluster_resource_classic_test.go

```diff
@@ -0,0 +1,210 @@
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	. "github.com/onsi/ginkgo/v2/dsl/core" // nolint
+	. "github.com/onsi/gomega"             // nolint
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/openshift-online/ocm-sdk-go/logging"
+)
+
+type MockHttpClient struct {
+	response *http.Response
+}
+
+func (c MockHttpClient) Get(url string) (resp *http.Response, err error) {
+	return c.response, nil
+}
+
+var (
+	clusterState      = &ClusterRosaClassicState{}
+	clusterId         = "1n2j3k4l5m6n7o8p9q0r"
+	clusterName       = "my-cluster"
+	regionId          = "us-east-1"
+	multiAz           = true
+	rosaCreatorArn    = "arn:aws:iam::123456789012:dummy/dummy"
+	apiUrl            = "https://api.my-cluster.com:6443"
+	consoleUrl        = "https://console.my-cluster.com"
+	machineType       = "m5.xlarge"
+	availabilityZone1 = "us-east-1a"
+	availabilityZone2 = "us-east-1b"
+	ccsEnabled        = true
+	awsAccountID      = "123456789012"
+	privateLink       = false
+	oidcEndpointUrl   = "example.com"
+	roleArn           = "arn:aws:iam::123456789012:role/role-name"
+	mockHttpClient    = MockHttpClient{
+		response: &http.Response{
+			TLS: &tls.ConnectionState{
+				PeerCertificates: []*x509.Certificate{
+					&x509.Certificate{
+						Raw: []byte("nonce"),
+					},
+				},
+			},
+		},
+	}
+)
+
+func generateBasicRosaClassicClusterJson() map[string]interface{} {
+	return map[string]interface{}{
+		"id": clusterId,
+		"region": map[string]interface{}{
+			"id": regionId,
+		},
+		"multi_az": multiAz,
+		"properties": map[string]interface{}{
+			"rosa_creator_arn": rosaCreatorArn,
+		},
+		"api": map[string]interface{}{
+			"url": apiUrl,
+		},
+		"console": map[string]interface{}{
+			"url": consoleUrl,
+		},
+		"nodes": map[string]interface{}{
+			"compute_machine_type": map[string]interface{}{
+				"id": machineType,
+			},
+			"availability_zones": []interface{}{
+				availabilityZone1,
+			},
+		},
+		"ccs": map[string]interface{}{
+			"enabled": ccsEnabled,
+		},
+		"aws": map[string]interface{}{
+			"account_id":   awsAccountID,
+			"private_link": privateLink,
+			"sts": map[string]interface{}{
+				"oidc_endpoint_url": oidcEndpointUrl,
+				"role_arn":          roleArn,
+			},
+		},
+	}
+}
+
+var _ = Describe("Rosa Classic cluster creation", func() {
```

Comment:
This is worng description, `populateRosaClassicClusterState` is not related only to the creation, it used by all crud functions